core/src/main/java/inetsoft/web/viewsheet/EventAspectService.java

```diff
@@ -71,6 +73,40 @@ public Void updateViewsheet(@ClusterProxyKey String runtimeId, CommandDispatcher
       return null;
    }
 
+   @ClusterProxyMethod(WorksheetEngine.CACHE_NAME)
+   public Void setWSExecution(@ClusterProxyKey String runtimeId, boolean undoable, Principal principal) throws Exception {
+      RuntimeWorksheet rws = viewsheetService.getWorksheet(runtimeId, principal);
+      MVSession session = rws.getAssetQuerySandbox().getMVSession();
+      WSExecution.setAssetQuerySandbox(rws.getAssetQuerySandbox());
```

Comment:
This doesn't work. This sets a thread local. It's highly likely that subsequent proxy method calls will be handled in a different thread.